openedx/core/djangoapps/programs/utils.py

```diff
@@ -105,3 +105,35 @@ def get_programs_for_dashboard(user, course_keys):
             log.exception('Unable to parse Programs API response: %r', program)
 
     return course_programs
+
+
+def get_programs_for_credentials(user, programs_credentials, credential_configuration):
+    """ Build a dictionary of programs
+
+    Given a user and an iterable of programs credentials, get all the programs
+    data and return it in a list of dictionaries.
```

Comment:
"""Given a user and an iterable of credentials, get corresponding programs data and return it as a list of dictionaries."""